app/src/main/java/com/github/studydistractor/sdp/createActivity/CreateActivityViewModel.kt

```diff
@@ -0,0 +1,41 @@
+package com.github.studydistractor.sdp.createActivity
+
+import com.github.studydistractor.sdp.ui.state.CreateActivityUiState
+import com.google.android.gms.tasks.Task
+import kotlinx.coroutines.flow.StateFlow
+
+interface CreateActivityViewModel {
```

Comment:
you could inherit from ViewModel() ?

Reply:
I'm not sure what change is requested here. If you mean that `CreateActivityViewModel` should inherit from `ViewModel()`, this is not possible because an interface cannot inherit from a class. However, both `CreateEventViewModel` and `CreateDistractionViewModel` inherit from `CreateActivityViewModel` and from `ViewModel()`, if that's what you have in mind.
E.g. `class CreateEventViewModel(
    createEventModel: CreateEventModel
) : CreateActivityViewModel, ViewModel()`

Reply:
Oh ! That's right, all good. I indeed meant that CreateActivityViewModel should inherit from ViewModel, but it's not possible

Reply:
Actually, maybe you can make the class abstract?